crypto/sm2/sm2_za.c

```diff
@@ -59,7 +65,7 @@ int sm2_compute_userid_digest(uint8_t *out,
         goto done;
     }
 
-    memset(out, 0, EVP_MD_size(digest));
+    memset(out, 0, hashlen);
```

Comment:
Side-note: seeing "hashlen" and "out" in the same line is a bit jarring.  I'd prefer hashlen were called outlen.

Reply:
Actually one can argue that this memset is redundant. EVP_Digest doesn't make assumption about out being zeroed. Secondly one can argues if subroutine returns error, then it shouldn't be writing to `out` at all. Caller should pay attention to return value and may not rely on it being cleared. And if this memset is redundant, then call to EVP_MD_size is redundant. Besides, this is private function, so we can look at caller, and caller checked EVP_MD_size and even zeroed out (out is zalloc-ed).

Reply:
I removed the memset.